thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/datalayer/bao/DAOTestBase.java

```diff
@@ -0,0 +1,279 @@
+import com.linkedin.thirdeye.datalayer.dto.EntityToEntityMappingDTO;
+import com.linkedin.thirdeye.datalayer.dto.JobDTO;
+import com.linkedin.thirdeye.datalayer.dto.MetricConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.OnboardDatasetMetricDTO;
+import com.linkedin.thirdeye.datalayer.dto.OverrideConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.RawAnomalyResultDTO;
+import com.linkedin.thirdeye.datalayer.pojo.AlertConfigBean;
+import com.linkedin.thirdeye.datalayer.util.DaoProviderUtil;
+import com.linkedin.thirdeye.datalayer.util.PersistenceConfig;
+import com.linkedin.thirdeye.datasource.DAORegistry;
+import com.linkedin.thirdeye.detector.email.filter.AlphaBetaAlertFilter;
+import com.linkedin.thirdeye.detector.metric.transfer.ScalingFactor;
+import com.linkedin.thirdeye.util.ThirdEyeUtils;
+
+import java.io.File;
+import java.io.FileReader;
+import java.net.URL;
+import java.sql.Connection;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.tomcat.jdbc.pool.DataSource;
+import org.joda.time.DateTime;
+
+public class DAOTestBase implements DaoProvider {
+  protected AnomalyFunctionManager anomalyFunctionDAO;
+  protected RawAnomalyResultManager rawAnomalyResultDAO;
+  protected JobManager jobDAO;
+  protected TaskManager taskDAO;
+  protected MergedAnomalyResultManager mergedAnomalyResultDAO;
+  protected DatasetConfigManager datasetConfigDAO;
+  protected MetricConfigManager metricConfigDAO;
+  protected DashboardConfigManager dashboardConfigDAO;
+  protected OverrideConfigManager overrideConfigDAO;
+  protected AlertConfigManager alertConfigDAO;
+  protected DataCompletenessConfigManager dataCompletenessConfigDAO;
+  protected EventManager eventDAO;
+  protected DetectionStatusManager detectionStatusDAO;
+  protected AutotuneConfigManager autotuneConfigDAO;
+  protected ClassificationConfigManager classificationConfigDAO;
+  protected EntityToEntityMappingManager entityToEntityMappingDAO;
+  protected GroupedAnomalyResultsManager groupedAnomalyResultsDAO;
+  protected OnboardDatasetMetricManager onboardDatasetMetricDAO;
+  protected ConfigManager configDAO;
+  protected ApplicationManager applicationDAO;
+
+  //  protected TestDBResources testDBResources;
+  protected DAORegistry daoRegistry;
+  DataSource ds;
+  String dbUrlId;
+
+  private static DaoProvider INSTANCE = new DAOTestBase();
+
+  private DAOTestBase(){
+    init();
+  }
+
+  public static DaoProvider getInstance(){
+    return INSTANCE;
+  }
+
+  protected void init() {
+    try {
+      URL url = TestDBResources.class.getResource("/persistence-local.yml");
+      File configFile = new File(url.toURI());
+      PersistenceConfig configuration = DaoProviderUtil.createConfiguration(configFile);
+      initializeDs(configuration);
+
+      DaoProviderUtil.init(ds);
+
+      daoRegistry = DAORegistry.getInstance();
```

Comment:
Can we do lazy initialization? For instance, we don't set DAO during init() but only when the corresponding get method is invoked. Because most tests only uses one or two of the DAOs.

Reply:
I am also thinking of this. Otherwise, it is expensive when a new test begins. The problem is the current function uses the create-schema.sql and drop-tables.sql to create and cleanup the whole DB. There is no information for us to construct a single table. We need to go through the sql file. I don't know if there is any other way to solve this.